<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Hugo 0.38.1" />


<title>RNA-seq Commands on MARCC - Practical Genomics Workshop</title>
<meta property="og:title" content="RNA-seq Commands on MARCC - Practical Genomics Workshop">



  








<link href='//cdn.bootcss.com/highlight.js/9.11.0/styles/github.min.css' rel='stylesheet' type='text/css' />



<link rel="stylesheet" href="/css/fonts.css" media="all">
<link rel="stylesheet" href="/css/main.css" media="all">



  </head>
  <body>
    <div class="wrapper">
      <header class="header">
        <nav class="nav">
  <a href="/" class="nav-logo">
    <img src="/images/wordlecircle_400x400.jpg"
         width="50"
         height="50"
         alt="Logo">
  </a>

  <ul class="nav-links">
    
    <li><a href="/preparation/">Preparation</a></li>
    
    <li><a href="/slack/">Slack</a></li>
    
    <li><a href="https://www.rstudio.com/resources/cheatsheets/">Cheatsheets</a></li>
    
    <li><a href="/rstudio/">RStudio</a></li>
    
    <li><a href="/marcc/">MARCC</a></li>
    
    <li><a href="http://genomics.jhu.edu/workshop">Registration</a></li>
    
  </ul>
</nav>

      </header>


<main class="content" role="main">

  <article class="article">
    

    <h1 class="article-title">RNA-seq Commands on MARCC</h1>

    

    <div class="article-content">
      <ul>
<li>RSEM 1.3.0</li>
<li>Bowtie 1.2.2</li>
<li>SAMtools 1.3</li>
</ul>
<pre><code>ssh gateway2.marcc.jhu.edu -l username@jhu.edu

srun -p shared -c 1 --mem 1024 -t 30 --pty bash

module load anaconda-python/3.6
source activate pg2018

cd ~/scratch
# mkdir genomes
cd genomes
# cp ~/work/genomes/chr20.fa.gz .
# gunzip chr20.fa.gz
cp ~/work/genomes/hg19-iGenomes.gtf .
cp ~/work/genomes/hg19.sizes .

module load bowtie/1.2.2
rsem-prepare-reference --bowtie --gtf hg19-iGenomes.gtf chr20.fa chr20rsem

cd ..
# cp -R ~/work/rawdata .
cd rawdata
# ./doRename.sh ALIAS
gunzip HUES64.mRNA*gz

cd ..
mkdir rna-seq
cd rna-seq

rsem-calculate-expression --output-genome-bam --paired-end --strandedness none ../rawdata/HUES64.mRNA.chr20.R1.fastq ../rawdata/HUES64.mRNA.chr20.R2.fastq ../genomes/chr20rsem HUES64_1

gunzip ~/scratch/rawdata/ectoderm.mRNA*gz
rsem-calculate-expression --output-genome-bam --paired-end --strandedness none ~/scratch/rawdata/ectoderm.mRNA.chr20.R1.fastq ~/scratch/rawdata/ectoderm.mRNA.chr20.R2.fastq ~/scratch/genomes/chr20rsem ectoderm_1

rsem-generate-data-matrix HUES64_1.genes.results ectoderm_1.genes.results &gt; HUES64_vs_ectoderm.genes.matrix
rsem-run-ebseq HUES64_vs_ectoderm.genes.matrix 1,1 HUES64_vs_ectoderm.genes.EBSeq
rsem-control-fdr HUES64_vs_ectoderm.genes.EBSeq 0.05 HUES64_vs_ectoderm.genes.DiffExp_fdr.txt

samtools sort -o HUES64_1.genome_sorted.bam HUES64_1.genome.bam
samtools index HUES64_1.genome_sorted.bam
rsem-bam2wig HUES64_1.genome_sorted.bam HUES64_1.genome.wig HUES64_1
wigToBigWig HUES64_1.genome.wig ../genomes/hg19.sizes HUES64_1.genome.bw

samtools sort -o ectoderm_1.genome_sorted.bam ectoderm_1.genome.bam
samtools index ectoderm_1.genome_sorted.bam
rsem-bam2wig ectoderm_1.genome_sorted.bam ectoderm_1.genome.wig ectoderm_1
wigToBigWig ectoderm_1.genome.wig ../genomes/hg19.sizes ectoderm_1.genome.bw</code></pre>

    </div>
  </article>

  


</main>

      <footer class="footer">
        <ul class="footer-links">
          <li>
            <a href="/index.xml" type="application/rss+xml" target="_blank">RSS feed</a>
          </li>
          <li>
            <a href="https://gohugo.io/" class="footer-links-kudos">Made with <img src="/images/hugo-logo.png" width="22" height="22"></a>
          </li>
        </ul>
      </footer>

    </div>
    



<script src="//cdn.bootcss.com/highlight.js/9.11.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/yaml.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



    

    
  </body>
</html>

